src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -903,6 +920,13 @@ async Task<(bool, ImmutableArray<ProjectConfiguration>)> CheckAsync(UpToDateChec
                     _lastFailureReason = logger.FailureReason ?? "";
                 }
             }
+
+            static async Task<(T1, T2)> WhenAll<T1, T2>(Task<T1> t1, Task<T2> t2)
+            {
+                await Task.WhenAll(t1, t2);
+
+                return (await t1, await t2);
```

Comment:
Since the tasks have already been completed, is the use of `await` here for each task to simply get the result value of each task?

Reply:
That's right. Could also write `.GetAwaiter().GetResult()` but I don't like seeing that in code.

I would like to pull this out into something a bit more generic and reusable, to make it easier to run multiple async operations concurrently. Just need to find the right API for that. I suspect it'd take a custom awaiter implementation.

Reply:
Discussion about this idea: https://github.com/dotnet/csharplang/discussions/380
And someone made a library to do something similar: https://github.com/buvinghausen/TaskTupleAwaiter/
Package on NuGet: https://www.nuget.org/packages/TaskTupleAwaiter/1.0.1

Reply:
Thanks. I've filed https://github.com/dotnet/project-system/pull/8223 to make this more generally reusable.